azurerm/internal/services/eventgrid/resource_arm_eventgrid_event_subscription.go

```diff
@@ -516,7 +744,135 @@ func expandEventGridEventSubscriptionFilter(d *schema.ResourceData) *eventgrid.E
 		filter.IsSubjectCaseSensitive = &caseSensitive
 	}
 
-	return filter
+	if advancedFilter, ok := d.GetOk("advanced_filter"); ok {
+		advancedFilters := make([]eventgrid.BasicAdvancedFilter, 0)
+		for _, v := range advancedFilter.([]interface{}) {
+			config := v.(map[string]interface{})
+
+			if filter, err := expandAdvancedFilter(config); err == nil {
+				advancedFilters = append(advancedFilters, filter)
+			} else {
+				return nil, err
+			}
+		}
+		filter.AdvancedFilters = &advancedFilters
+	}
+
+	return filter, nil
+}
+
+func expandAdvancedFilter(config map[string]interface{}) (eventgrid.BasicAdvancedFilter, error) {
+	operatorType := config["operator_type"].(string)
+	key := config["key"].(string)
+	value := config["value"].(string)
+	values := utils.ExpandStringSlice(config["values"].([]interface{}))
+
+	switch operatorType {
+	case string(eventgrid.OperatorTypeAdvancedFilter),
+		string(eventgrid.OperatorTypeBoolEquals),
+		string(eventgrid.OperatorTypeNumberGreaterThan),
+		string(eventgrid.OperatorTypeNumberGreaterThanOrEquals),
+		string(eventgrid.OperatorTypeNumberLessThan),
+		string(eventgrid.OperatorTypeNumberLessThanOrEquals):
+		if values != nil && len(*values) > 0 {
```

Comment:
add comment that this is a workaround as long as nested schema validation is not working as expected. see lines 295 and 303.

Reply:
done